File: java/pl.jakubwojt.kontomierz.test.client/src/pl/jakubwojt/kontomierz/test/client/ClientServiceFactory.java
package pl.jakubwojt.kontomierz.test.client;

import pl.jakubwojt.kontomierz.test.client.services.CachingLayer;
import pl.jakubwojt.kontomierz.test.client.services.TransportLayer;
import pl.jakubwojt.kontomierz.test.client.services.ValidationLayer;
import pl.jakubwojt.kontomierz.test.common.factory.ApplicationServiceFactory;
import pl.jakubwojt.kontomierz.test.common.services.layers.base.ApplicationService;

public class ClientServiceFactory implements ApplicationServiceFactory 
{
	public ApplicationService getApplicationService() {
		return new CachingLayer(new ValidationLayer(new TransportLayer()));
	}
}

File: cs/Layers.Client/ClientServiceFactory.cs
using Layers.Client.Services.Layers;
using Layers.Common.Factory;
using Layers.Common.Services;
using Layers.Common.Services.Layers.Logging;
using Layers.Common.Services.Layers.Validation;

namespace Layers.Client
{
    public class ClientServiceFactory: IServiceFactory
    {
        private static readonly IServicesLayer ApplicationStack;
        private static readonly ILoggingLayer LoggingLayer;
        static ClientServiceFactory()
        {
            var tmpAppStack  =  new ValidationLayer(new LocalStorageLayer(new TransportLayer()));
            ApplicationStack = (IServicesLayer) (LoggingLayer = new LoggingLayer(tmpAppStack));
        }

        public IBasketService GetBasketService()
        {
            return ApplicationStack;
        }

        public IShopService GetShopService()
        {
            return ApplicationStack;
        }

        public IShopAdminService GetShopAdminService()
        {
            return ApplicationStack;
        }

        public IUserService GetUserService()
        {
            return ApplicationStack;
        }

        public ILoggingLayer GetLoggingLayer()
        {
            return LoggingLayer;
        }

    }
}
